### PR TITLE
[dv/lc_ctrl] testplan and add to V1 status

### PR DESCRIPTION
**Pull request by cindychip**
This PR has three commits to move LC_CTRL to V1 status:
1. Add a testplan.
2. Remove the interrupts and memory tests within testplan and lc_ctrl_sim_cfg.hjson
3. Update checklist.

### CODE REVIEW

**Review by sriyerg — APPROVED** [review]
LGTM, thanks!

File: hw/ip/lc_ctrl/data/lc_ctrl_testplan.hjson [hunk 1]
```diff
@@ -3,30 +3,123 @@
...
+            - Initialize lc_ctrl by sending pwrmgr req and otp_ctrl valid random data
+            - Request a valid next LC state by writing CSRs: `transition_target`,
+              `transition_token*`, and `transition_cmd`
 
             **Checks**:
-            - TBD
+            - After lc_ctrl initialization finishes, check lc_ctrl broadcast outputs, check
+              the value of lc_state and lc_transition_cnt CSRs, and check device_id and id_state
+              CSRs
+            - During lc_ctrl state transition request, check if `transition_regwen` register is set
```
Comment (weicaiyang) [inline hw/ip/lc_ctrl/data/lc_ctrl_testplan.hjson:24]: I feel it's better to test this in a separate test as it's very timing sensitive. I have a sequence to test this kind of reg. FYI
https://github.com/lowRISC/opentitan/blob/master/hw/ip/keymgr/dv/env/seq_lib/keymgr_cfgen_vseq.sv

**Review by weicaiyang — APPROVED** [review]
LGTM with a suggestion

File: hw/ip/lc_ctrl/data/lc_ctrl_testplan.hjson (hunk 1 above)
Comment (cindychip) [reply hw/ip/lc_ctrl/data/lc_ctrl_testplan.hjson:24]: Thank you for your suggestion and reference. I will separate it as you recommended.

**Review by udinator — APPROVED** [review]

File: hw/ip/lc_ctrl/data/lc_ctrl_testplan.hjson [hunk 2]
```diff
@@ -12,21 +11,135 @@
...
+      name: jtag_priority
+      desc: '''
+            This test covers a corner case in JTAG and TLUL interfaces.
+
+            **Stimulus**:
+            - Issue transition requests from TLUL and JTAG interfaces at the same time
+
+            **Checks**:
+            - Ensure TAP interface has the priority
+            - Once request is completed, ensure the non-prioritized interface's CSR reading are all
```
Comment (msfschaffner) [inline hw/ip/lc_ctrl/data/lc_ctrl_testplan.hjson:138]: It's not entirely clear what the "request" refers to here, so it may make sense to clarify this part. I think we'd want to ensure that right after the "mutex claim" operation, the non-prioritized interface returns 0 from the CSRs.

File: hw/ip/lc_ctrl/doc/checklist.md [hunk 3]
```diff
@@ -115,28 +115,28 @@ Review        | Signoff date            | Not Started |
...
+Tests         | [SIM_SMOKE_TEST_PASSING][]            | Done        |
+Tests         | [SIM_CSR_MEM_TEST_SUITE_PASSING][]    | Done        |
+Tests         | [FPV_MAIN_ASSERTIONS_PROVEN][]        | N/A         |
+Tool Setup    | [SIM_ALT_TOOL_SETUP][]                | Done        |
+Regression    | [SIM_SMOKE_REGRESSION_SETUP][]        | Done        |
+Regression    | [SIM_NIGHTLY_REGRESSION_SETUP][]      | Done        |
+Regression    | [FPV_REGRESSION_SETUP][]              | N/A         |
+Coverage      | [SIM_COVERAGE_MODEL_ADDED][]          | Done        |
+Code Quality  | [TB_LINT_SETUP][]                     | Done        |
+Integration   | [PRE_VERIFIED_SUB_MODULES_V1][]       | N/A         | Exception for IP modules
```
Comment (msfschaffner) [inline hw/ip/lc_ctrl/doc/checklist.md:135]: Do we typically enumerate which submodules are excepted?

**Review by msfschaffner — APPROVED** [review]
LGTM with two comments.

File: hw/ip/lc_ctrl/data/lc_ctrl_testplan.hjson (hunk 2 above)
Comment (cindychip) [reply hw/ip/lc_ctrl/data/lc_ctrl_testplan.hjson:138]: Good point Michael, let me amend that.

File: hw/ip/lc_ctrl/doc/checklist.md (hunk 3 above)
Comment (cindychip) [reply hw/ip/lc_ctrl/doc/checklist.md:135]: Are you thinking about modules like prim_alert_sender, prim_lc_sync, etc? Here we are referring to if we excluded any sub-modules from coverage result. Currently we did not exclude anything from coverage. This is mainly used for top-level where there are lots of pre-verified IPs.

Comment (msfschaffner) [reply hw/ip/lc_ctrl/doc/checklist.md:135]: Ah ok I see. Sorry I misread that.

Comment (cindychip) [reply hw/ip/lc_ctrl/doc/checklist.md:135]: No worries at all. That was a good point tho :)
